### PR TITLE
WTF-16574 Add jquery adapter.

### PR DESCRIPTION
**Pull request by djvirgen**
## Usage:

1. Add `jquery` and `monocle-client-jquery-min.js` to your page.
2. Start making calls!

```js
$.monocle.get('/users/123')
.then(function(user) {
  // got user!
})
.catch(function(error) {
  // error :(
});
```

The base path may be set via `$.monocle.setBasePath('/my-api')`

### CODE REVIEW

File: test/unit/lib/http_adapter/jquery_test.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+                        .then(function(result) {
+                            this.$.ajax.lastCall.args[0].should.have.property('timeout', 5000);
+                        }.bind(this));
+                    });
+                });
+
+                describe('headers', function() {
+                    it('passes content-type header', function() {
+                        var expectedContentType = 'application/json';
+                        return this.adapter.request(method, this.path, this.options)
```
Comment (eviechan) [inline test/unit/lib/http_adapter/jquery_test.js]: Where is ```this.options``` set?  Maybe I'm just missing the reference somewhere.

Also it'd be nice to see that ```data``` attribute is set properly in the ajax call so we can see that options argument is tested properly in the call to ```this.$.ajax```.

Comment (djvirgen) [inline test/unit/lib/http_adapter/jquery_test.js]: Both of these are now addressed.

File: test/unit/lib/http_adapter/jquery_test.js [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+        it('sets data according in XHR request', function() {
+            this.options.body = {
+                foo: 'test foo'
+            };
+            return this.adapter.request('post', '/foo', this.options)
+            .then(function(result) {
+                this.$.ajax.calledOnce.should.be.true;
+                this.$.ajax.lastCall.args[0].should.have.property('data');
+                this.$.ajax.lastCall.args[0].data.should.equal(this.options.body);
```
Comment (eviechan) [inline test/unit/lib/http_adapter/jquery_test.js]: :boom:

**Comment by eviechan** [comment]
:cherry_blossom:

**Pull request merged**